src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -389,9 +404,113 @@ public class SystemNamespaceSymbol : NamespaceSymbol
+            if (!SyntaxTreeGroupingBuilder.ValidateFilePath(filePath, out var validateFilePathFailureBuilder))
+            {
+                context.Diagnostics.Write(validateFilePathFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(context.Arguments[0].syntax)));
+                return null;
+            }
+
+            var fileUri = context.Binder.FileResolver.TryResolveFilePath(context.Binder.FileUri, filePath);
+            if (fileUri is null)
+            {
+                context.Diagnostics.Write(DiagnosticBuilder.ForPosition(context.Arguments[0].syntax).FilePathCouldNotBeResolved(filePath, context.Binder.FileUri.LocalPath));
+                return null;
+            }
+
+            if (!fileUri.IsFile)
+            {
+                context.Diagnostics.Write(DiagnosticBuilder.ForPosition(context.Arguments[0].syntax).UnableToLoadNonFileUri(fileUri));
+                return null;
+            }
+            return fileUri;
+        }
+
+        private static TypeSymbol LoadTextContentTypeBuilder(FunctionOverload.AdvancedReturnTypeBuilderContext context)
+        {
+            var fileUri = GetFileUriWithDiagnostics(context);
+            if (fileUri is null)
+            {
+                return LanguageConstants.String;
+            }
+
+            var fileEncoding = Encoding.Default;
+            if (context.Arguments.Length > 1)
+            {
+                fileEncoding = Encoding.UTF8; //todo read encoding
+            }
+
+            if (!context.Binder.FileResolver.TryRead(fileUri, out var fileContent, out var fileReadFailureBuilder, fileEncoding, LanguageConstants.MaxLiteralCharacterLimit))
+            {
+                context.Diagnostics.Write(fileReadFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(context.Arguments[0].syntax)));
+                return LanguageConstants.String;
+            }
+
+            return new StringLiteralType(fileContent);
+        }
+
+        private static TypeSymbol LoadContentAsBase64TypeBuilder(FunctionOverload.AdvancedReturnTypeBuilderContext context)
+        {
+            var fileUri = GetFileUriWithDiagnostics(context);
+            if (fileUri is null)
+            {
+                return LanguageConstants.String;
+            }
+
+            if (!context.Binder.FileResolver.TryReadAsBase64(fileUri, out var fileContent, out var fileReadFailureBuilder, LanguageConstants.MaxLiteralCharacterLimit))
+            {
+                context.Diagnostics.Write(fileReadFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(context.Arguments[0].syntax)));
+                return LanguageConstants.String;
+            }
+
+            return new StringLiteralType(fileContent);
+        }
+
+        private static bool StringLiteralFunctionReturnTypeEmitter(JsonTextWriter writer, EmitterContext emitterContext, FunctionCallSyntax functionCallSyntax, FunctionSymbol functionSymbol, out SyntaxBase? nextExpression, out LanguageExpression languageExpression)
```

Comment:
>StringLiteralFunctionReturnTypeEmitter [](http://example.com/codeflow?start=28&length=38)

We wanted to keep emit code out of the semantic model so it's easier to swap out that layer. I'm guessing this is just here because the PR is still in progress?

Reply:
Yes, exactly. 

I looked at the `DecoratorEvaluator` and I'll implement it in a similar fashion. My idea is to create a delegate that will transform transform `functionCallSyntax` into `SyntaxBase` (in this case - `StringLiteralSyntax`) and in that form pass to the emitter/expression converter for transforming. 

WDYT?